src/ctap/mod.rs

```diff
@@ -466,7 +469,10 @@ impl<E: Env> StatefulPermission<E> {
     }
 
     /// Returns the next storage credential key for enumeration and advances it.
-    pub fn next_enumerate_credential(&mut self) -> Result<usize, Ctap2StatusCode> {
+    pub fn next_enumerate_credential(&mut self, env: &mut E) -> Result<usize, Ctap2StatusCode> {
+        if env.clock().is_elapsed(&self.permission) {
+            self.clear();
+        }
```

Comment:
Look like those 3 lines could be their own function. Namely, making `clear_timer` private instead of deleting it, and calling it instead of inlining it.

Reply:
Done